src/types/reducers.js

```diff
@@ -78,6 +122,7 @@ export type UrlState = {
   threadOrder: ThreadIndex[],
   hiddenThreads: ThreadIndex[],
   transforms: TransformStacksPerThread,
+  zipFilePath: string | null,
```

Comment:
The duplication is necessary because they represent two different things. The `UrlState` is the "desired path in the zip file to view" while the `AppState` (now the `ZippedProfilesState`) represent the current async state of getting to the desired `UrlState`.

For updating based on URL changes, this handled in the React component that I haven't published as part of this URL. When the component mounts, it kicks off all the async behavior, and then it checks for `UrlState` changes when receiving new props.